### DISCUSSION TITLE
is `Astro.request` available in components?

### DISCUSSION THREAD

**Discussion by mayank99** [lilnasy/gratelets#92] [Q&A]
Considering that `Astro.request` is not even available in Astro components, I'd assume that it's also not made available by the [`global`](https://github.com/lilnasy/gratelets/tree/main/packages/global) package?

**Comment by lilnasy** [comment]
It is available!

...same as Astro.request is available in Astro components. Are you referring to the fact that response headers are read-only in astro components?

**Reply by mayank99** [reply]
I was actually trying to access `Astro.request.formData()`, but yes response headers would be another case too. Basically, I want to access all the things that a page would have access to.

**Reply by lilnasy** [reply]
It should have all properties except `Astro.slots`. You can read the body using formdata() provided that the request body has not already been consumed by text() or arrayBuffer() elsewhere.

It doesn't have `Astro.slots` because it powered by a middleware under the hood which only has access to the `APIContext`.
